test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.js\"");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFile(cwd, "jquery.js", caseInsensitive: true);
+            LibmanTestsUtility.WaitForRestoredFileNotPresent(cwd, "jquery.min.js", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Unpkg_AddNewLibraryWithSpecifingFiles()
+        {
+            _libManConfig.Open();
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "unpkg", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("unpkg");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"LICENSE.txt\"");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFile(cwd, "LICENSE.txt", caseInsensitive: true);
+            LibmanTestsUtility.WaitForRestoredFileNotPresent(cwd, "dist", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Cdnjs_DeleteLibrary()
```

Comment:
>FileSaveRestore_Cdnjs_DeleteLibrary() [](start = 20, length = 37)

Using another existing project with a ready libman.json can reduced the amount of Editor typing related code here.

Reply:
yes it could, that's a good idea.

---
In reply to: [194350175](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194350175) [](ancestors = 194350175)

Reply:
I could probably try getting content from the other json file as you point out for reverting file change.

---
In reply to: [194552391](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194552391) [](ancestors = 194552391,194350175)